src/containerapp/azext_containerapp/_utils.py

```diff
@@ -1714,3 +1727,192 @@ def format_location(location=None):
     if location:
         return location.lower().replace(" ", "").replace("(", "").replace(")", "")
     return location
+
+
+def is_docker_running():
+    # check to see if docker is running
+    client = None
+    out = True
+    try:
+        client = docker.from_env()
+        # need any command that will show the docker daemon is not running
+        client.containers.list()
+    except docker.errors.DockerException:
+        out = False
+    finally:
+        if client:
+            client.close()
+    return out
+
+
+def get_pack_exec_path():
+    try:
+        dir_path = os.path.dirname(os.path.realpath(__file__))
+        bin_folder = os.path.join(dir_path, "bin")
+        if not os.path.exists(bin_folder):
+            os.makedirs(bin_folder)
+
+        pack_cli_version = "v0.29.0"
+        exec_name = "pack"
+        compressed_download_file_name = f"pack-{pack_cli_version}"
+        host_os = platform.system()
+        if host_os == "Windows":
+            compressed_download_file_name = f"{compressed_download_file_name}-windows.zip"
+            exec_name = "pack.exe"
+        elif host_os == "Linux":
+            compressed_download_file_name = f"{compressed_download_file_name}-linux.tgz"
+        elif host_os == "Darwin":
+            compressed_download_file_name = f"{compressed_download_file_name}-macos.tgz"
+        else:
+            raise Exception(f"Unsupported host OS: {host_os}")
+
+        exec_path = os.path.join(bin_folder, exec_name)
+        if os.path.exists(exec_path):
```

Comment:
Just want to double check on the expected behavior here. Are we gonna install pack CLI no matter if user has already installed pack CLI and added to their PATH? If not, we could check using `os.environ["PATH"]`

Reply:
Correct, we aren't going to make any assumptions about what the user has installed on their machine -- this allows us to force the `containerapp` module to use a specific version of the pack CLI that we choose and _know_ will work with the code that is released with each module version moving forward